pkg/reader/ccip.go

```
@@ -1294,65 +1294,23 @@ func (r *ccipChainReader) getOffRampSourceChainsConfig(
-	contractBatch := make(types.ContractBatch, 0, len(chains))
-	sourceChains := make([]any, 0, len(chains))
-
-	for _, chain := range chains {
-		if chain == r.destChain {
-			continue
-		}
-		sourceChains = append(sourceChains, chain)
-
-		contractBatch = append(contractBatch, types.BatchRead{
-			ReadName: consts.MethodNameGetSourceChainConfig,
-			Params: map[string]any{
-				"sourceChainSelector": chain,
-			},
-			ReturnVal: new(SourceChainConfig),
-		})
-	}
-
-	results, _, err := r.contractReaders[r.destChain].ExtendedBatchGetLatestValues(
-		ctx, contractreader.ExtendedBatchGetLatestValuesRequest{consts.ContractNameOffRamp: contractBatch},
-		false,
-	)
-
+	// Use the ConfigPoller to handle caching
+	configs, err := r.configPoller.GetSourceChainConfigs(ctx, r.destChain, chains)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get source chain configs for dest chain %d: %w",
-			r.destChain, err)
+		return nil, fmt.Errorf("get source chain configs: %w", err)
 	}
 
-	lggr.Debugw("got source chain configs", "configs", results)
-
-	// Populate the map.
-	for _, readResult := range results {
-		if len(readResult) != len(sourceChains) {
-			return nil, fmt.Errorf("selectors and source chain configs length mismatch: sourceChains=%v, configs=%v",
-				sourceChains, results)
-		}
-		for i, chainSel := range sourceChains {
-			v, err := readResult[i].GetResult()
-			if err != nil {
-				return nil, fmt.Errorf("GetSourceChainConfig for chainSelector=%d failed: %w", chainSel, err)
-			}
-
-			cfg, ok := v.(*SourceChainConfig)
-			if !ok {
-				return nil, fmt.Errorf("invalid result type from GetSourceChainConfig for chainSelector=%d: %w", chainSel, err)
-			}
-
+	// Filter out disabled chains if needed
+	if !includeDisabled {
+		for chain, cfg := range configs {
 			enabled, err := cfg.check()
-			if err != nil {
-				return nil, fmt.Errorf("source chain config check for chain %d failed: %w", chainSel, err)
+			if err != nil || !enabled {
```

is it safe to ignore the err case here?

---

I'm not sure but it's preferable to stick to the original behavior. Good catch, thanks.